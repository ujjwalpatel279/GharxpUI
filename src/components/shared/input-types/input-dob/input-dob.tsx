import React, { FunctionComponent, useEffect, useState } from 'react';
import { useForm } from 'react-hook-form/dist/index.ie11';
import { validation } from '../../../../shared/validation/validation';
import { Input } from '..';

interface InputDobProps {
  register: ReturnType<typeof useForm>['register'];
  required: boolean;
  errors?: ReturnType<typeof useForm>['errors'];
  setValue?: ReturnType<typeof useForm>['setValue'];
  value?: string;
  name: string;
  getValues?: ReturnType<typeof useForm>['getValues'];
  disabled?: boolean;
}

export const InputDob: FunctionComponent<InputDobProps> = ({
  register,
  required,
  errors,
  setValue,
  value,
  name,
  disabled,
}) => {
  const [day, setDay] = useState<string>();
  const [month, setMonth] = useState<string>();
  const [year, setYear] = useState<string>();

  useEffect(() => {
    if (value) {
      const date = value.split('-');
      setDay(date[2].split('T')[0]);
      setMonth(date[1]);
      setYear(date[0]);
    }
  }, [value]);

  return (
    <div className="l-flex">
      <div>
        <Input
          id={`${name}Day`}
          error={errors && !!errors[`${name}Day`]}
          placeHolder="DD"
          required={required}
          ref={register(validation.dob.types.day)}
          setValue={setValue}
          value={day}
          disabled={disabled}
        />
      </div>
      <div>
        <Input
          id={`${name}Month`}
          error={errors && !!errors[`${name}Month`]}
          placeHolder="MM"
          required={required}
          ref={register(validation.dob.types.month)}
          setValue={setValue}
          value={month}
          disabled={disabled}
        />
      </div>
      <div>
        <Input
          id={`${name}Year`}
          error={errors && !!errors[`${name}Year`]}
          placeHolder="YYYY"
          required={required}
          ref={register(validation.dob.types.year)}
          setValue={setValue}
          value={year}
          disabled={disabled}
        />
      </div>
    </div>
  );
};
