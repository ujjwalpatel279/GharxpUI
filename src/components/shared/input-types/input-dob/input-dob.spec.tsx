import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { InputDob } from './input-dob';

describe('Input DOB Component', () => {
  afterEach(cleanup);

  const mock = jest.fn();

  it('renders the DOB component', () => {
    const { container } = render(<InputDob register={mock} required={true} name="dob" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
