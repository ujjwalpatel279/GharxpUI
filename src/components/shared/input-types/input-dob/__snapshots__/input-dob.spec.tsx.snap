// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input DOB Component renders the DOB component 1`] = `
<div
  class="l-flex"
>
  <div>
    <input
      aria-invalid="false"
      class="c-input isRequired"
      id="dobDay"
      name="dobDay"
      placeholder="DD"
      required=""
      type="text"
    />
  </div>
  <div>
    <input
      aria-invalid="false"
      class="c-input isRequired"
      id="dobMonth"
      name="dobMonth"
      placeholder="MM"
      required=""
      type="text"
    />
  </div>
  <div>
    <input
      aria-invalid="false"
      class="c-input isRequired"
      id="dobYear"
      name="dobYear"
      placeholder="YYYY"
      required=""
      type="text"
    />
  </div>
</div>
`;
